### DISCUSSION TITLE
An option for the browser field to handle a one to many relationship

### DISCUSSION THREAD

**Discussion by LiquidPL** [area17/twill#853] [Q&A] [answered]
I'm trying to use a browser field to set a one-to-many relationship between two of my models, however I've noticed that it's only supporting many-to-many fields currently. 

Are there any plans to add one-to-many support? If not, would you be willing to accept PRs on that? I would be interested in making one.

**Comment by yanhao-li** [comment, marked as answer]
Hi @LiquidPL ,
Yes, unfortunately, the browser field currently only supports many-to-many relationship. If you really want to use one-to-many relationship, we do have a workaround with the following code in the repository (replacing the `browserName` and `browserModel` etc):

```php
public function prepareFieldsBeforeSave($object, $fields)
    {
        $fields['belongsToModel_id'] = array_get($fields, 'browsers.browserName.0.id', null);

        return parent::prepareFieldsBeforeSave($object, $fields);
    }

    public function getFormFields($object)
    {
        $fields = parent::getFormFields($object);

        $belongsToModel = $object->belongsToModel;

        if ($belongsToModel) {
            $fields['browsers']['browserName'] = collect([
                [
                    'id' => $belongsToModel->id,
                    'name' => $belongsToModel->title,
                    'edit' => moduleRoute('belongsToModuleName', 'routePrefix', 'edit', $belongsToModel->id),
                    'thumbnail' => $belongsToModel->defaultCmsImage(['w' => 100, 'h' => 100]),

                ],
            ])->toArray();
        }

        return $fields;
    }
```

and of course, a PR for adding support for this feature will be very welcomed! Please let us know if you need any help.
